views/pages/editTask.ejs

```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <%- include('../partials/head'); %>
+</head>
+
+<body class="container">
+
+  <header>
+    <%- include('../partials/header'); %>
+  </header>
+
+  <main>
+    <p>
+      <%= message %>
+    </p>
+    <form action=<%= "/tasks/update/" + task.id %> method="post">
+      <label for="name">Name:</label> <br>
+      <input name="name" value="<%=task.name %>"><br>
+      <label for="complete">Completed: </label>
+      <% if (task.completed) { %>
+        <input type="checkbox" name="complete" value="true" checked><br><br>
+        <% } else { %>
+          <input type="checkbox" name="complete" value="false"><br><br>
+          <% } %>
```

Comment:
@akosasante I see, now it makes much more sence, thank you! 
Another issue that I´ve noticed (maybe it is connected with this matter)
when I create/ edit task I can see in payload 'complete: true' only when the checkbox was marked; BUT if the checkbox was not checked this fiend doesn´t appear in payload in the console 
![image](https://github.com/NatachaKey/ejs-demo/assets/103755090/acb7d60c-81a0-4c14-9731-accf18468e68)
![image](https://github.com/NatachaKey/ejs-demo/assets/103755090/7a3e03f3-944e-42b1-a0ed-d6199a2daf1c)

it seems like we don´t actually pass complete:false (if the checkbox was not marked) to the server- just to make sure- does this happen by default or we should change something in the code to make it work correctly?

Reply:
This isn't something we can change, in this case. This is how HTML forms interact with the browser. That's why in the `updateTask` we have to write code for what to do if `req.body.complete` is not passed in and thus is null/undefined (because the input wasn't checked). 

If we wrote client-side code (eg: if we rendered the HTML and had a `<script>` tag or a separate JS file. Or if we had a front-end framework like React or Vue), then we could use `fetch` or some other similar method to write AJAX code. In that case, we could define how we want to send the information in the form the the backend. And we could then write code to actually send `complete: false` if the checkbox is not checked.